File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/ServerTypes/SignLettersDB/SignLettersColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SignLettersRow } from "./SignLettersRow";

export interface SignLettersColumns {
    SignTitle: Column<SignLettersRow>;
    CreatedDate: Column<SignLettersRow>;
}

export class SignLettersColumns extends ColumnsBase<SignLettersRow> {
    static readonly columnsKey = 'SignLettersDB.SignLetters';
    static readonly Fields = fieldsProxy<SignLettersColumns>();
}

File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/SignLettersDB/SignLetters/SignLettersColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace CorrespondenceSystem.SignLettersDB.Columns;

[ColumnsScript("SignLettersDB.SignLetters")]
[BasedOnRow(typeof(SignLettersRow), CheckNames = true)]
public class SignLettersColumns
{
    [DisplayName("Title")]    
    public string SignTitle { get; set; }
    public DateTime CreatedDate { get; set; } //= DateTime.Now;

}